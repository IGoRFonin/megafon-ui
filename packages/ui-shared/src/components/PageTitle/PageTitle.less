@import '~@megafon/ui-core/styles/base.less';

@b: e('.mfui-beta-page-title');

@{b} {
    &__inner {
        position: relative;
        padding-top: 100px;

        @media @mobileB {
            padding-top: 90px;
        }

        @media @mobileSM {
            padding-top: 45px;
        }
    }

    &__breadcrumbs {
        position: absolute;
        top: 32px;
        left: 0;

        @media @mobileSM {
            display: none;
        }
    }

    &__badge {
        .small();
        position: absolute;
        top: 75px;
        left: 0;
        padding: 2px 14px;
        font-weight: 500;
        background-color: @purple;
        color: @clearWhite;

        @media @desktopS {
            top: 67px;
        }

        @media @mobileB {
            top: 59px;
        }

        @media @mobileSM {
            top: 16px;
        }
    }
}
