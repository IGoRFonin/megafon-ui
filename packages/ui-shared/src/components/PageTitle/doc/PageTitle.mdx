---
name: PageTitle
menu: Конструктор
route: components/page-title
---

import Example from './PageTitle.example.mdx';
import Props from './PageTitle.props.mdx';

# VideoBanner

<CopyToClipboardBox text="import { PageTitle } from '@megafon/ui-shared';" />

<DoczTabs>
    <Example />
    <Props />
</DoczTabs>
