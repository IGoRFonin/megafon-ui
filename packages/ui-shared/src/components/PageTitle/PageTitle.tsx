import React from 'react';
import PropTypes from 'prop-types';
import { cnCreate, Header } from '@megafon/ui-core';
import './PageTitle.less';
import Breadcrumbs, { Props as BreadcrumbsType } from '../Breadcrumbs/Breadcrumbs';

type Props = {
    /** Текст заголовка */
    title: string;
    /** Хлебные крошки */
    breadcrumbs?: BreadcrumbsType['items'];
    /** Текст бейджа */
    badge?: string;
};

const cn = cnCreate('mfui-beta-page-title');
const PageTitle: React.FC<Props> = ({ title, breadcrumbs, badge }) => (
    <div className={cn()}>
        <div className={cn('inner')}>
            {breadcrumbs?.length && <Breadcrumbs items={breadcrumbs} className={cn('breadcrumbs')} />}
            {badge && <div className={cn('badge')}>{badge}</div>}
            <Header>{title}</Header>
        </div>
    </div>
);

PageTitle.propTypes = {
    title: PropTypes.string.isRequired,
    breadcrumbs: PropTypes.arrayOf(
        PropTypes.shape({
            title: PropTypes.string.isRequired,
            href: PropTypes.string,
        }).isRequired
    ),
    badge: PropTypes.string,
};

export default PageTitle;
