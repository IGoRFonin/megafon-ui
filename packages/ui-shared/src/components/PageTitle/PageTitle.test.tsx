import React from 'react';
import { shallow } from 'enzyme';
import PageTitle from './PageTitle';

const breadcrumbs = [
    {
        title: 'Тарифы',
        href: '#',
    },
    {
        title: 'Управляй !',
        href: '#',
    },
    {
        title: 'Менеджер',
        href: '#',
    },
];

const badge = 'badge';

describe('PageTitle', () => {
    it('should render component', () => {
        const wrapper = shallow(<PageTitle title="title" />);

        expect(wrapper).toMatchSnapshot();
    });

    it('should render with optional props', () => {
        const wrapper = shallow(<PageTitle title="title" breadcrumbs={breadcrumbs} badge={badge} />);

        expect(wrapper).toMatchSnapshot();
    });
});
