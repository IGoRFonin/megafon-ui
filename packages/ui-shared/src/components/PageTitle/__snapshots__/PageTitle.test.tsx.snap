// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageTitle should render component 1`] = `
<div
  className="mfui-beta-page-title"
>
  <div
    className="mfui-beta-page-title__inner"
  >
    <Header
      as="h1"
      color="black"
      hAlign="inherit"
    >
      title
    </Header>
  </div>
</div>
`;

exports[`PageTitle should render with optional props 1`] = `
<div
  className="mfui-beta-page-title"
>
  <div
    className="mfui-beta-page-title__inner"
  >
    <Breadcrumbs
      className="mfui-beta-page-title__breadcrumbs"
      items={
        Array [
          Object {
            "href": "#",
            "title": "Тарифы",
          },
          Object {
            "href": "#",
            "title": "Управляй !",
          },
          Object {
            "href": "#",
            "title": "Менеджер",
          },
        ]
      }
    />
    <div
      className="mfui-beta-page-title__badge"
    >
      badge
    </div>
    <Header
      as="h1"
      color="black"
      hAlign="inherit"
    >
      title
    </Header>
  </div>
</div>
`;
